### DISCUSSION TITLE
About --disable-writeback-cache options

### DISCUSSION THREAD

**Discussion by M-S4TO** [Azure/azure-storage-fuse#1456] [Q&A] [answered]
### Question  
  
Hello community,  
  
I am currently using Blobfuse2 to mount Azure Blob Storage. I have come across several documents with the following comment:  
 https://github.com/Azure/azure-storage-fuse/blob/f088b803fe387bbb1f5f76caedbe75cf2439b003/setup/baseConfig.yaml#L16-L19
  
What I would like to know is **whether setting the `disable-writeback-cache` option to `true` makes writes to Blob Storage bypass Blobfuse2 and directly write the data.**  
  
I am not very familiar with Unix or FUSE, so at first, I imagined that this flag was related to userland programs writing to a cache SSD mounted on a VM. However, as I looked into the code, I started to doubt my initial understanding.  
  
Specifically, I would like to know which data's writeback cache is disabled when enabling the `disable-writeback-cache` option. I am also interested in how the behavior of writes to Blob Storage changes with this option enabled.  
  
Thank you very much for your assistance.  
  
Best regards,

**Comment by bhansalivikas** [comment, marked as answer]
"write-back cache" is more inclined to kernel page caching behavior and has nothing to do with how Blobfuse upstream the data to storage container. You can use block-cache mode to upstream data to container as your write (there will be some buffering at client level and a periodic staging of the blocks). However, actual persistence will happen only when you close the file handle or fsync the data.

**Reply by M-S4TO** [reply]
Thank you very much for your detailed explanation regarding the "write-back cache" and its behavior with Blobfuse2. Your response clarified my doubts and provided the information I needed. I appreciate your assistance!

Best regards,
